scripts/quests/windurst/Carbuncle_Debacle.lua

```diff
@@ -0,0 +1,186 @@
+        {
+            ['Koru-Moru'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 1 then
+                        return quest:event(416)
+                    elseif questProgress == 4 then
+                        return quest:progressEvent(417)
+                    elseif questProgress == 5 then
+                        return quest:progressEvent(418)
+                    elseif questProgress == 7 then
+                        return quest:progressEvent(419)
+                    elseif player:getQuestStatus(xi.quest.log_id.WINDURST, xi.quest.id.windurst.CARBUNCLE_DEBACLE) == QUEST_COMPLETED then
+                        return player:event(420)
+                    end
+                end,
+            },
+                onEventFinish =
+                {
+                    [416] = function(player, csid, option, npc)
+                        quest:setVar(player, 'Prog', 2)
+                    end,
+
+                    [417] = function(player, csid, option, npc)
+                        quest:setVar(player, 'Prog', 5)
+                        npcUtil.giveKeyItem(player, xi.ki.DAZE_BREAKER_CHARM)
+                    end,
+
+                    [419] = function(player, csid, option, npc)
+                        quest:complete(player)
+                        player:addTitle(xi.title.PARAGON_OF_SUMMONER_EXCELLENCE)
+                    end
+                }
+            },
+
+        [xi.zone.MHAURA] =
+        {
+
+            ['Ripapa'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 2 then
+                        return quest:progressEvent(10022)
+                    elseif questProgress == 3 and not player:hasItem(xi.items.LIGHTNING_PENDULUM) then
+                        return quest:event(10023)
+                    end
+                end,
+            },
+
+                onEventFinish =
+                {
+                    [10022] = function(player, csid, option, npc)
+                        npcUtil.giveItem(player, xi.items.LIGHTNING_PENDULUM)
+                        quest:setVar(player, 'Prog', 3)
+                    end,
+
+                    [10023] = function(player, csid, option, npc)
+                        npcUtil.giveItem(player, xi.items.LIGHTNING_PENDULUM)
+                    end
+                }
+
+        },
+
+        [xi.zone.CLOISTER_OF_STORMS] =
+        {
+            onEventFinish =
+            {
+                [32001] = function(player, csid, option, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+                    if player:getLocalVar('battlefieldWin') == 577 and questProgress == 3 then
+                        quest:setVar(player, 'Prog', 4)
+                    end
```

Comment:
I believe event 32001 is the win cutscene, so you should only need to check for quest prog below.
Moreover, we don't need to load questprog into a local var if we only access it the one time.
```suggestion
                [32001] = function(player, csid, option, npc)
                    if quest:getVar(player, 'Prog') == 3 then
                        quest:setVar(player, 'Prog', 4)
                    end
```